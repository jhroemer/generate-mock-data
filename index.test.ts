import path from "node:path";
import { readFile, readdir } from "node:fs/promises";
import { execa } from "execa";
import { expect, test } from "vitest";

const directory = process.cwd();
const outputDir = "./output";

async function getFixtureFileContent() {
  const fixtureDir = await readdir(path.join(directory, outputDir));
  const unprocessedFixtureFile = await readFile(
    path.join(directory, outputDir, fixtureDir[0])
  );
  const fixtureFile = unprocessedFixtureFile.toString();

  return fixtureFile;
}

test("Generate data", async () => {
  const subprocess = execa`node dist/cli.js --output-dir output`;

  for await (const line of subprocess) {
    if (line.includes("Enter URL")) {
      subprocess.stdin?.write("http://localhost:9090/hello-world\r");
      subprocess.stdin?.end();
    }
  }

  const fixtureFileContent = await getFixtureFileContent();
  expect(fixtureFileContent).toMatchSnapshot();
});
