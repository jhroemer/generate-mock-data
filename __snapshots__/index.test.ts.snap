// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generate data 1`] = `
"const GET_http_localhost_9090_hello_world_123 = {
  method: "GET",
  url: "http://localhost:9090/hello-world",
  statusCode: 200,
  response: [{ hello: "world" }],
};

export default GET_http_localhost_9090_hello_world_123;
"
`;
